packages/wpcom.js/src/lib/site.media.js

```diff
@@ -153,8 +152,12 @@ Media.prototype.addFiles = function ( query, files, fn ) {
 
 	const videoFiles = this.filterFilesUploadableOnVideoPress( files );
 	if ( videoFiles.length ) {
-		const uploader = new TusUploader( this.wpcom, this._sid );
-		return uploader.startUpload( videoFiles );
+		return import( /* webpackChunkName: "wpcom-tus-uploader" */ './tus-uploader' ).then(
+			( { default: TusUploader } ) => {
+				const uploader = new TusUploader( this.wpcom, this._sid );
+				return uploader.startUpload( videoFiles );
+			}
+		);
```

Comment:
This requires that the consumer of the package has a way to interpret and perform dynamic imports. That's a quite big ask for some build environments.

It would be better to completely remove this code from the `wpcom` library and move it to the place where it's used. There is probably only one, in the Calypso Media Library. This has been already discussed in https://github.com/Automattic/wp-calypso/pull/62100#issuecomment-1541889708